camkes/parser/fdtQueryEngine.py

```diff
@@ -413,6 +414,133 @@ class DtbMatchQuery(Query):
     def __init__(self):
         self.engine = None
 
+    @staticmethod
+    def unpack(data, sizes):
+        result = []
+        while sizes:
+            sz = sizes.pop(0)
+            result.append(functools.reduce(lambda a, b: (a << 32) + b, data[0:sz]))
+            data = data[sz:]
+        return result, data
+
+    @staticmethod
+    def unpacker(data, sizes):
+        while data:
+            result, data = DtbMatchQuery.unpack(data, sizes[:])
+            yield result
+
+    @staticmethod
+    def pack(x, size):
+        result = []
+        if size > 1:
+            result = DtbMatchQuery.pack(x >> 32, size - 1)
+        else:
+            if x >= (1 << 32):
+                raise RuntimeError('value 0x%x does not fit into device tree' % (x))
+        result.append(x & ((1 << 32) - 1))
+        return result
+
+    @staticmethod
+    def xlat_reg(addr, size, buses):
+        for bus in buses:
+            if not 'translations' in bus:
+                continue
+            for xlat in bus['translations']:
+                if addr < xlat['from_start'] or addr > xlat['from_start'] + xlat['size'] - 1:
+                    continue
+                if addr + size > xlat['from_start'] + xlat['size']:
+                    raise RuntimeError('Region 0x%x/0x%x does not fit into range 0x%x/0x%x'
+                        % (addr, size, xlat['from_start'], xlat['size']))
+                    continue
+                addr = xlat['to_start'] + (addr - xlat['from_start'])
+                break
+        return addr
+
+    @staticmethod
+    def xlat_regs(cells):
+        if not 'reg' in cells:
+            return
+
+        this_address_cells = cells['buses'][0]['this-address-cells']
```

Comment:
check first that `cells['buses'][0]` exists?

Reply:
See my comment below.

Reply:
ok, maybe add a comment in the source code. To reduce redundancy consider using:
```
bus0 = cells['buses'][0] # guaranteed to exists
this_address_cells = bus0['this-address-cells']
this_size_cells = bus0['this-size-cells']
```

Reply:
See above.